core/tauri/src/api/file/extract.rs

```diff
@@ -119,9 +119,31 @@ impl<'a> Extract<'a> {
         let mut archive = zip::ZipArchive::new(source)?;
         for i in 0..archive.len() {
           let mut file = archive.by_index(i)?;
-          let path = into_dir.join(file.name());
-          let mut output = fs::File::create(path)?;
-          io::copy(&mut file, &mut output)?;
+          // Decode the file name from raw bytes instead of using file.name() directly.
+          // file.name() uses String::from_utf8_lossy() which may return messy characters
+          // such as: τê▒Σ║ñµÿô.app/, that does not work as expected.
+          // Here we require the file name must be a valid UTF-8.
+          let file_name = String::from_utf8(file.name_raw().to_vec())?;
+          let out_path = into_dir.join(&file_name);
+          if file_name.ends_with('/') || file_name.ends_with('\\') {
```

Comment:
can't you just check `file.is_dir()` instead ?

Reply:
They are basically the same in this case, because both / and \ are ASCII characters. file.is_dir() uses file.name() internally, here we check it ourself just wants the code logic be consistent.

Reply:
If that is the case then `file.is_dir()` is better to use here, it is more concise.

Reply:
Done.